tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -98,6 +98,8 @@ sub run {
     record_info 'Resource Group', "Resource Group used for deployment: $deployment_name";
 
     my $provider = $self->provider_factory();
+    set_var('SLES4SAP_SSHKEY', get_ssh_private_key_path());
```

Comment:
Two things:
1) This could be `PUBLIC_CLOUD_SSHKEY`
2) Please don't forget to mention this variable in `variables.md`

Reply:
unless it is really required let's try to keep **common** codebase and this would be easier with common variables

Reply:
It is not really a variable, this is something specific about this test module and how it handles the YAML  templates for the `qe-sap-deployment`. The underlay mechanism expect that anything in the YAML template like `%SOMETHING%` is replaced by value of a setting named `SOMETHING`. In this specific case, and in others like `VNET_ADDRESS_RANGE`, `SUBNET_ADDRESS_RANGE`, `HANA_TOKEN` ..., the value cannot be provided as a setting from the JobGroup or others. On the other hand value has to be calculated during the test, setting a variable like `set_vars` is only a way to later inject this in the YAML template resolution mechanism. 
As you can see, only `set_vars` is used and no any `get_vars`

Reply:
It is still a variable.